internal/component/prometheus/scrape/scrape.go

```diff
@@ -335,7 +336,9 @@ func (c *Component) Run(ctx context.Context) error {
 			// Prometheus handles marking series as stale: it is the client's responsibility to inject the
 			// staleness markers. In our case, for targets that moved to another instance in the cluster, we hand
 			// over this responsibility to the new owning instance. We must not inject staleness marker here.
-			c.scraper.DisableEndOfRunStalenessMarkers(jobName, movedTargets)
+			if len(movedTargets) > 0 {
```

Comment:
Why not just have `DisableEndOfRunStalenessMarkers` immediately return when the length is 0, prior to attempting to lock a mutex?

Reply:
Yeah, I can do that some time later because it'd require changing the fork and I don't have capacity to do that right now.

Reply:
I can do it for you and update your branch?

Reply:
sounds good!

Reply:
Done!

Reply:
Thanks so much for the help!